File: 228. Summary Ranges/main.py
from typing import List


class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        result = list()
        size =len(nums)
        i = 0

        while i < size:
            start = nums[i]

            while i + 1 < size and nums[i + 1] == nums[i] + 1:
                i += 1

            if start == nums[i]:
                result.append(str(start))
            else:
                result.append(str(start) + "->" + str(nums[i]))

            i += 1

        return result

class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        result = list()
        size =len(nums)
        i = 0

        while i < size:
            start = nums[i]

            while i + 1 < size and nums[i + 1] == nums[i] + 1:
                i += 1

            if start == nums[i]:
                result.append(f"{start}")
            else:
                result.append(f"{start}" + "->" + f"{nums[i]}")

            i += 1

        return result

File: 228. Summary Ranges/main.rs
struct Solution;

impl Solution {
    pub fn summary_ranges(nums: Vec<i32>) -> Vec<String> {
        let mut result = Vec::new();
        let mut i = 0;
        let size = nums.len();

        while i < size {
            let start = nums[i];

            while i + 1 < size && nums[i + 1] == nums[i] + 1 {
                i += 1;
            }

            if start == nums[i] {
                result.push(start.to_string());
            } else {
                result.push(start.to_string() + "->" + &nums[i].to_string());
            }

            i += 1;
        }

        result
    }
}

impl Solution {
    pub fn summary_ranges(nums: Vec<i32>) -> Vec<String> {
        let mut result = Vec::new();
        let mut i = 0;
        let size = nums.len();

        while i < size {
            let start = nums[i];

            while i + 1 < size && nums[i + 1] == nums[i] + 1 {
                i += 1;
            }

            if start == nums[i] {
                result.push(format!("{}", start));
            } else {
                result.push(format!("{}->{}", start, nums[i]));
            }

            i += 1;
        }

        result
    }
}
